simulation/core/yaml_processing.py

```diff
@@ -0,0 +1,187 @@
+    else:
+        raise ValueError('Must specify either filepath or yaml_dict.')
+
+    # Get the number of devices (GPUs)
+    if 'compute' in config and 'gpus' in config['compute']:
+        total_devices = int(config['compute']['gpus'])
+    else:
+        total_devices = None
+
+    workers = None
+    train_dataset = None
+    max_duration = None
+    global_batch_size = None
+    # Get the training and dataset params
+    if 'parameters' in config:
+        config = om.create(config['parameters'])
+
+    om.resolve(config)
+
+    assert isinstance(config, DictConfig), 'config must be a dict.'
+
+    # get global batch size
+    if 'global_train_batch_size' in config:
+        global_batch_size = config['global_train_batch_size']
+    elif 'batch_size' in config:
+        global_batch_size = config['batch_size']
+
+    # get number of workers and training dataset params
+    if 'train_loader' in config:
+        train_loader = config['train_loader']
+        if 'num_workers' in train_loader:
+            workers = train_loader['num_workers']
+        else:
+            workers = 1
+        if 'dataset' in train_loader:
+            train_dataset = train_loader['dataset']
+        else:
+            raise ValueError('dataset must be specified in the yaml file.')
+    elif 'dataset' in config:
+        dataset = config['dataset']
+        if 'train_dataset' in dataset:
+            train_dataset = dataset['train_dataset']
+            if 'streaming_kwargs' in train_dataset:
+                # Merge streaming kwargs, if present, into train_dataset
+                train_dataset.update(train_dataset['streaming_kwargs'])
+            if 'dataloader_kwargs' in train_dataset and 'num_workers' in train_dataset[
+                    'dataloader_kwargs']:
+                workers = train_dataset['dataloader_kwargs']['num_workers']
+            else:
+                workers = 1
+        else:
+            raise ValueError('train_dataset must be specified in the yaml file.')
+    else:
+        raise ValueError('train_loader or dataset must be specified in the yaml file.')
+
+    # Get duration of training from config
+    if 'max_duration' in config:
+        max_duration = config['max_duration']
+    elif 'trainer' in config and 'max_duration' in config['trainer']:
+        max_duration = config['trainer']['max_duration']
+    else:
+        raise ValueError('max_duration must be specified in the yaml file.')
+
+    # convert max_duration to epochs or batches.
+    max_duration = ensure_time(max_duration, TimeUnit.EPOCH)
+    time_unit = max_duration.unit
+    if time_unit != TimeUnit.EPOCH and time_unit != TimeUnit.BATCH:
+        raise ValueError('Simulator currently only supports max_duration in epochs or batches.')
+
+    # convert train_dataset to dict, if it isn't already
+    if isinstance(train_dataset, DictConfig):
+        train_dataset = om.to_container(train_dataset)
+
+    assert isinstance(workers, int), 'workers must be an integer.'
```

Comment:
I would recommend raising a `ValueError` instead of an `assert` statement in the source code.

Reply:
done for all lines!